app.R

```diff
@@ -280,40 +280,37 @@ render_map <- function(subexp_name, input, output, full_cache_data) {
 }
 
 # render outputs for a given subexperiment
-render_subexp_output <- function(subexp_name, input, output, full_cache_data) {
+render_subexp_output <- function(subexp_name, exp_name, input, output, full_cache_data) {
   
-  render_variable_menu(subexp_name, output, full_cache_data)
+  render_variable_menu(subexp_name, exp_name, output, full_cache_data)
   
-  render_cultivar_menu(subexp_name, input, output, full_cache_data)
+  render_cultivar_menu(subexp_name, exp_name, input, output, full_cache_data)
   
-  render_trait_plot(subexp_name, input, output, full_cache_data)
+  render_trait_plot(subexp_name, exp_name, input, output, full_cache_data)
   
-  render_plot_hover(subexp_name, input, output, full_cache_data)
+  render_plot_hover(subexp_name, exp_name, input, output, full_cache_data)
   
   if (!is.null(full_cache_data[[ subexp_name ]][[ 'managements' ]])) {
     
-    render_mgmt_timeline(subexp_name, input, output, full_cache_data)
+    render_mgmt_timeline(subexp_name, exp_name, input, output, full_cache_data)
     
-    render_timeline_hover(subexp_name, input, output, full_cache_data)
+    render_timeline_hover(subexp_name, exp_name, input, output, full_cache_data)
     
   }
 
-  render_map(subexp_name, input, output, full_cache_data)
+  render_map(subexp_name, exp_name, input, output, full_cache_data)
 }
 
 render_experiment_output <- function(experiment_name, input, output, full_cache_data) {
-  lapply(names(full_cache_data[[ experiment_name ]]), render_subexp_output, input, output, full_cache_data[[ experiment_name ]])
+  lapply(names(full_cache_data[[ experiment_name ]]), render_subexp_output, experiment_name, input, output, full_cache_data[[ experiment_name ]])
```

Comment:
Why is full_cache_data indexed by experiment_name here but subexp_name above?

Reply:
`full_cache_data` is indexed by `subexp_name` in the `render_subexp_output` function above instead of `experiment_name` since a subsetted version of `full_cache_data` is passed into the `render_subexp_output` function. The `full_cache_data` object in the environment of the `render_subexp_output` function therefore only contains data for a single experiment and can only be subsetted by sub experiment names (not experiment names).

The `render_experiment_output` function loops through all sub experiments for an experiment and runs the `render_subexp_output` on each sub experiment. `render_suebxp_output` then generates the output for a single sub experiment of the experiment (eg. Biomass is a sub experiment of MAC Season 1).